core/app/login/token/[token]/route.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * This route is used to accept customer login token JWTs from the
+ * [Customer Login API](https://developer.bigcommerce.com/docs/start/authentication/customer-login)
+ * and log the customers in using alternative authentication methods
+ */
+
+import { decodeJwt } from 'jose';
+import { isRedirectError } from 'next/dist/client/components/redirect-error';
+// eslint-disable-next-line @typescript-eslint/no-restricted-imports
+import { redirect, unstable_rethrow as rethrow } from 'next/navigation';
+
+import { signIn } from '~/auth';
+
+export async function GET(request: Request, { params }: { params: Promise<{ token: string }> }) {
+  const token = (await params).token;
+
+  // check if token is valid
+  try {
+    decodeJwt(token); // throws error if token is an invalid JWT
+
+    // decode token without checking signature to get redirect path
+    // token is not checked for validity here, so it could be expired or invalid at this point
+    // token validity is checked in the signIn function
+    const claims = decodeJwt(token);
+    const redirectTo = claims.redirect_to ? (claims.redirect_to as string) : '/account/orders';
+    const channelId = claims.channel_id?.toString() || process.env.BIGCOMMERCE_CHANNEL_ID;
+
+    // sign in with token which will check validity against BigCommerce API
+    // and redirect to redirectTo
+    await signIn('credentials', {
+      type: 'jwt',
+      jwt: token,
+      redirectTo,
+      channelId,
+    });
+  } catch (error) {
+    if (isRedirectError(error)) {
+      rethrow(error);
+    }
```

Comment:
Rethrow should handle the redirect error.

```suggestion
    rethrow(error);
```